### PR TITLE
Ensure Jabbax and Plug are loaded before defining a module in which they're required

### PR DESCRIPTION
**Pull request by andreyuhai**
Even though this compiles fine on its own, when you depend on Surgex, you can't compile it because you get

```
❯ mix deps.compile surgex
==> surgex
Compiling 47 files (.ex)

== Compilation error in file lib/surgex/parser/parser.ex ==
** (CompileError) lib/surgex/parser/parser.ex:67: Jabbax.Document.Resource.__struct__/0 is undefined, cannot expand struct Jabbax.Document.Resource. Make sure the struct name is correct. If the struct name exists and is correct but it still cannot be found, you likely have cyclic module usage in your code
    expanding struct: Jabbax.Document.Resource.__struct__/0
    lib/surgex/parser/parser.ex:67: Surgex.Parser.parse/2
could not compile dependency :surgex, "mix compile" failed. Errors may have been logged above. You can recompile this dependency with "mix deps.compile surgex", update it with "mix deps.update surgex" or clean it with "mix deps.clean surgex"
```

```
❯ mix deps.compile surgex
==> surgex
Compiling 47 files (.ex)

== Compilation error in file lib/surgex/sentry/sentry.ex ==
** (CompileError) lib/surgex/sentry/sentry.ex:85: Plug.Conn.__struct__/0 is undefined, cannot expand struct Plug.Conn. Make sure the struct name is correct. If the struct name exists and is correct but it
 still cannot be found, you likely have cyclic module usage in your code
    expanding struct: Plug.Conn.__struct__/0
    lib/surgex/sentry/sentry.ex:85: Surgex.Sentry.scrub_params/1
could not compile dependency :surgex, "mix compile" failed. Errors may have been logged above. You can recompile this dependency with "mix deps.compile surgex", update it with "mix deps.update surgex" or
clean it with "mix deps.clean surgex"
```

### How to reproduce?

1. Just create a new app

```
mix new foobar --sup
```

2. Then add surgex as a dependency to your `mix.exs`

```
  defp deps do
    [
      {:surgex, "~> 4.7"},
    ]
  end
```

3. Run `mix deps.compile surgex` and you'll get the compilation errors I've listed above.

So in those files mentioned in the errors, we're still depending on the structs that come from those packages ([1](https://github.com/surgeventures/surgex/blob/5806b48b9503f35ea97f37bf8c322a1990884848/lib/surgex/parser/parser.ex#L67-L73), [2](https://github.com/surgeventures/surgex/blob/5806b48b9503f35ea97f37bf8c322a1990884848/lib/surgex/sentry/sentry.ex#L85)). When it's marked as optional, however those packages are not loaded.

For `optional` [the documentation says](https://hexdocs.pm/mix/1.12/Mix.Tasks.Deps.html#module-dependency-definition-options)

> `:optional` - marks the dependency as optional. In such cases, the current project will always include the optional dependency but any other project that depends on the current project won't be forced to use the optional dependency. However, if the other project includes the optional dependency on its own, the requirements and options specified here will also be applied. Optional dependencies will not be started by the application.

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -57,7 +57,9 @@ defmodule Surgex.Mixfile do
       {:ex_doc, "~> 0.23", only: :dev, runtime: false},
       {:ex_machina, "~> 2.3", only: :test},
       {:inch_ex, "~> 2.0", only: [:dev, :test]},
+      {:jabbax, "~> 1.0"},
```
Comment (kamilkowalski) [inline mix.exs:60]: Surgex is a mixed bag of helper modules - some aren't related to web APIs at all. A common pattern with optional dependencies is to only define the modules that require them if they're available - using [Code.ensure_loaded?/1](https://hexdocs.pm/elixir/1.15.5/Code.html#ensure_loaded?/1). I think that would be a better approach as it allows for using `surgex` without requiring and compiling `jabbax` if it's not necessary.

Comment (kamilkowalski) [reply mix.exs:60]: We actually use that approach in `Surgex.DateTime` and `Surgex.Changeset`.

Comment (andreyuhai) [reply mix.exs]: Okay, so I'm now using `Code.ensure_loaded?` as well. I've also bumped the version and updated the CHANGELOG.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,15 @@
...
 ## [Unreleased]
 
+## [4.15.1]
+
+-  Make sure
+
+    1. Jabbax is loaded before we define `Surgex.Parser`
+    2. Plug is loaded before we define `Surgex.Sentry`
+
+to avoid compilation issues in apps that depend on Surgex.
```
Comment (kamilkowalski) [inline CHANGELOG.md:12]: I know the function is `ensure_loaded?/1` but it's mostly used to verify if a module is available, so I'd reword this changelog:

```suggestion
-  Fix optional dependency on `Jabbax` and `Plug` by:

    1. defining `Surgex.Parser` only if `Jabbax` is available
    2. defining `Surgex.Sentry` only if `Plug` is available
```

**Review by kamilkowalski — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (andreyuhai) [reply CHANGELOG.md:12]: Wouldn't it try to load them if it's not there first anyway?
But the version you suggested makes more sense and looks better, so thank you!

**Review by kamilkowalski — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (kamilkowalski) [reply CHANGELOG.md]: @andreyuhai yes, it tries to load them if not loaded yet, but in the end returns a boolean saying if it's available or not.

Comment (andreyuhai) [reply CHANGELOG.md]: Yeap, got it, thank you! 🙇
